src/GraphQL/Resolvers/MemberResolver.cs

```diff
@@ -0,0 +1,168 @@
+            if (methodArgumentExpressions == null)
+                throw new ArgumentNullException(nameof(methodArgumentExpressions));
+            // verify that the expressions provided match the number of parameters
+            var methodParameters = methodInfo.GetParameters();
+            if (methodArgumentExpressions.Count != methodParameters.Length)
+            {
+                throw new InvalidOperationException("The number of expressions must equal the number of method parameters.");
+            }
+            if (instanceExpression.Parameters.Count != 1 ||
+                instanceExpression.Parameters[0].Type != typeof(IResolveFieldContext) ||
+                !methodInfo.DeclaringType!.IsAssignableFrom(instanceExpression.ReturnType))
+            {
+                throw new ArgumentException($"Source lambda must be of type Func<IResolveFieldContext, {methodInfo.DeclaringType!.Name}>.", nameof(instanceExpression));
+            }
+
+            // create a parameter expression for IResolveFieldContext
+            var resolveFieldContextParameter = Expression.Parameter(typeof(IResolveFieldContext), "context");
+
+            // convert each of the provided lambda expressions and pull the expression bodies, replacing the existing lambda's
+            // parameter with the new one, so each expression is using the same parameter
+            var expressionBodies = new Expression[methodArgumentExpressions.Count];
+            for (int i = 0; i < methodArgumentExpressions.Count; i++)
+            {
+                var expr = methodArgumentExpressions[i];
+                if (expr.Parameters.Count != 1 || expr.Parameters[0].Type != typeof(IResolveFieldContext) || expr.ReturnType != methodParameters[i].ParameterType)
+                {
+                    throw new InvalidOperationException($"A supplied expression is not a lambda delegate of type Func<IResolveFieldContext, {methodParameters[i].ParameterType.Name}>.");
+                }
+                expressionBodies[i] = expr.Body.Replace(expr.Parameters[0], resolveFieldContextParameter);
+            }
+
+            // create the method call expression
+            var methodCallExpr =
+                Expression.Call(
+                    methodInfo.IsStatic
+                        ? null
+                        : instanceExpression.Body.Replace(
+                            instanceExpression.Parameters[0],
+                            resolveFieldContextParameter),
+                    methodInfo,
+                    expressionBodies);
+
+            _resolver = BuildFieldResolver(resolveFieldContextParameter, methodCallExpr);
+        }
+
+        /// <summary>
+        /// Creates an appropriate resolver function based on the return type of the expression body.
+        /// </summary>
+        protected virtual Func<IResolveFieldContext, object?> BuildFieldResolver(ParameterExpression resolveFieldContextParameter, Expression bodyExpression)
```

Comment:
The contents of this method will expand drastically once we implement ValueTask.  You can see a prototype of the code in the draft PR #2907 now.  Regardless, this protected method is overridden now by the EventStreamMethodResolver.